packages/snaps-sdk/src/jsx/validation.ts

```diff
@@ -143,11 +143,34 @@ function element<Name extends string, Props extends ObjectSchema = EmptyObject>(
   });
 }
 
+/**
+ * A struct for the {@link ImageElement} type.
+ */
+export const ImageStruct: Describe<ImageElement> = element('Image', {
+  src: svg(),
+  alt: optional(string()),
+});
+
+const IconNameStruct: Struct<`${IconName}`, null> = nullUnion(
+  Object.values(IconName).map((name) => literal(name)) as any,
+);
+
+/**
+ * A struct for the {@link IconElement} type.
+ */
+export const IconStruct: Describe<IconElement> = element('Icon', {
+  name: IconNameStruct,
+  color: optional(
+    nullUnion([literal('default'), literal('primary'), literal('muted')]),
+  ),
+  size: optional(nullUnion([literal('md'), literal('inherit')])),
+});
+
 /**
  * A struct for the {@link ButtonElement} type.
  */
 export const ButtonStruct: Describe<ButtonElement> = element('Button', {
-  children: StringElementStruct,
+  children: nullUnion([StringElementStruct, ImageStruct, IconStruct]),
```

Comment:
Hmm, I'm unsure whether this should be using `children()`. WDYT @Mrtenz?

Reply:
It should.